src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        MachineProducer<? extends T> machineProducer;
+        if (type == Bulldozer.class) {
+            machineProducer = (MachineProducer<? extends T>) new BulldozerProducer();
+        } else if (type == Excavator.class) {
+            machineProducer = (MachineProducer<? extends T>) new ExcavatorProducer();
```

Comment:
Similar to line 15, this unchecked cast could lead to runtime issues. Ensure that the type safety is maintained.